multisense_ros/cfg/multisense.cfg

```diff
@@ -2,297 +2,343 @@
+            gen.add("ground_surface_pointcloud_global_min_z_m", double_t, 0, "Minimum pointcloud range for B-spline fit", 0.5, 0.0, 50.0)
+            gen.add("ground_surface_pointcloud_global_max_x_m", double_t, 0, "Maximum pointcloud width (LHS distance) for B-spline fit", 25.0, 0.0, 50.0)
+            gen.add("ground_surface_pointcloud_global_min_x_m", double_t, 0, "Minimum pointcloud width (LHS distance) for B-spline fit", -25.0, -50.0, 0.0)
+            gen.add("ground_surface_pointcloud_global_max_height_m", double_t, 0, "Maximum pointcloud height for B-spline fit", 10.0, 0.0, 50.0)
+            gen.add("ground_surface_pointcloud_global_min_height_m", double_t, 0, "Minimum pointcloud height for B-spline fit", -10.0, -50.0, 0.0)
+            gen.add("ground_surface_spline_resolution_x", int_t, 0, "Number of spline levels in the left camera optical frame's x dimension", 4, 0, 10)
+            gen.add("ground_surface_spline_resolution_z", int_t, 0, "Number of spline levels in the left camera optical frame's z dimension", 4, 0, 10)
+            base_model_enum = gen.enum([gen.const("Quadratic", str_t, "Quadratic", "Quadratic"),
+                                gen.const("Mean", str_t, "Mean", "Mean"),
+                                gen.const("Zero", str_t, "Zero", "Zero")],
+                                "Available base model settings");
+            gen.add("ground_surface_pre_transform_data", str_t, 0, "Model to transform raw data with before performing B-Spline fit", "Mean", edit_method=base_model_enum);
+
+            gen.add("ground_surface_pointcloud_grid_size", double_t, 0, "Size of XZ cells which the B-Spline is modelled  upon — larger cell sizes result in faster processing and coarser models", 0.5, 0.01, 1.0)
+            gen.add("ground_surface_min_points_per_grid", int_t, 0, "For each XZ cell column, there must be at least this many points in order to use the centroid in the B-spline modelling process — used to filter extranoues & noisy points from influencing B-spline model", 10, 0, 100)
+            gen.add("ground_surface_obstacle_height_thresh_m", double_t, 0, "Obstacle height in meters", 2.0, 0.0, 10.0)
+            gen.add("ground_surface_obstacle_percentage_thresh", double_t, 0, "Percentage of points in XZ cell column which must be above obstacle height threshold to consider the cell an obstacle", 0.50, 0.0, 1.0)
+            gen.add("ground_surface_max_fitting_iterations", int_t, 0, "ADVANCED SETTING: Number of iterations in B-spline routine", 10, 1, 30)
+            gen.add("ground_surface_adjacent_cell_search_size_m", double_t, 0, "ADVANCED SETTING: Border size around obstacle cells to examine during iterative B-spline routine", 1.5, 0.0, 5.0)
+            gen.add("ground_surface_spline_draw_resolution", double_t, 0, "Resolution to draw resulting B-Spline model with in RVIZ", 0.1, 0.01, 1.0)
+        #endif
```

Comment:
Here's where we (also) generate a different config as we iterate over the supported features